pkg/scheduler/core/generic_scheduler.go

```diff
@@ -312,9 +319,16 @@ func (g *genericScheduler) findNodesThatPassFilters(ctx context.Context, fwk fra
 			}
 		} else {
 			if !status.IsSuccess() {
-				statusesLock.Lock()
-				statuses[nodeInfo.Node().Name] = status
-				statusesLock.Unlock()
+				length := atomic.AddInt32(&errorNodesLen, 1)
+				if length > numNodesToFind {
```

Comment:
this is not correct, the loop could examine more than `numNodesToFind` number of nodes.